django/db/models/functions/datetime.py

```diff
@@ -183,17 +179,18 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
             raise ValueError('output_field must be either DateField, TimeField, or DateTimeField')
         # Passing dates or times to functions expecting datetimes is most
         # likely a mistake.
-        output_field = copy.output_field
-        explicit_output_field = field.__class__ != copy.output_field.__class__
+        class_output_field = self.__class__.output_field if isinstance(self.__class__.output_field, Field) else None
+        output_field = class_output_field or copy.output_field
+        has_explicit_output_field = class_output_field or type(field) is not type(copy.output_field)
```

Comment:
This required a bit of adjustments as _explicit_ output field can be now be defined at the class level as well. This to allow `TruncDate('time_field', output_field=TimeField())` to report `Cannot truncate TimeField 'time_field' to DateField` instead of `Cannot truncate TimeField 'time_field' to TimeField`.